packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -413,158 +412,63 @@ class AndroidSdk {
 
   /// Returns the version of java in the format \d(.\d)+(.\d)+
   /// Returns null if version not found.
-  String? getJavaVersion({
-    required AndroidStudio? androidStudio,
-    required FileSystem fileSystem,
-    required OperatingSystemUtils operatingSystemUtils,
-    required Platform platform,
-    required ProcessUtils processUtils,
-  }) {
-    final String? javaBinary = findJavaBinary(
-      androidStudio: androidStudio,
-      fileSystem: fileSystem,
-      operatingSystemUtils: operatingSystemUtils,
-      platform: platform,
-    );
-    if (javaBinary == null) {
-      globals.printTrace('Could not find java binary to get version.');
-      return null;
-    }
-    final RunResult result = processUtils.runSync(
-      <String>[javaBinary, '--version'],
-      environment: sdkManagerEnv,
-    );
-    if (result.exitCode != 0) {
-      globals.printTrace(
-          'java --version failed: exitCode: ${result.exitCode} stdout: ${result.stdout} stderr: ${result.stderr}');
-      return null;
-    }
-    return parseJavaVersion(result.stdout);
-  }
-
-  /// Extracts JDK version from the output of java --version.
-  @visibleForTesting
-  static String? parseJavaVersion(String rawVersionOutput) {
-    // The contents that matter come in the format '11.0.18' or '1.8.0_202'.
-    final RegExp jdkVersionRegex = RegExp(r'\d+\.\d+(\.\d+(?:_\d+)?)?');
-    final Iterable<RegExpMatch> matches =
-        jdkVersionRegex.allMatches(rawVersionOutput);
-    if (matches.isEmpty) {
-      globals.logger.printWarning(_formatJavaVersionWarning(rawVersionOutput));
-      return null;
-    }
-    final String? versionString = matches.first.group(0);
-    if (versionString == null || versionString.split('_').isEmpty) {
-      globals.logger.printWarning(_formatJavaVersionWarning(rawVersionOutput));
-      return null;
-    }
-    // Trim away _d+ from versions 1.8 and below.
-    return versionString.split('_').first;
+  ///
+  /// Callers should instead use the [Java] class.
+  // TODO(andrewkolos): see https://github.com/flutter/flutter/issues/126126.
+  String? getJavaVersion() {
+    return globals.java?.getVersion()?.shortText;
```

Comment:
Why would `globals.java` be null?

Reply:
Are there more callers of this than would be practical to migrate in one PR? I'm always worried about these kinds of migrations being left unfinished (such as `stringsArg` -> `stringsArgDeprecated`).

Reply:
Yeah that was part of the idea. With this change, I wanted to avoid changes to logic and focus on mostly refactoring logic out of `AndroidSdk`. I can do an all-in-one PR, but the diff would probably be around 400 lines larger (see my first unfinished draft: https://github.com/flutter/flutter/pull/126029).

Reply:
I'll experiment with migrating existing callers (without adding new callers) and see if the diff is still manageable

Reply:
> I'll experiment with migrating existing callers (without adding new callers, which is ultimately needed to fix the parent issue) and see if the diff is still manageable

Yeah, this sounds reasonable to me.

Reply:
> Why would `globals.java` be null?

Just realized I never answered this question. Across the codebase, a `null` indicates that no installation of Java was found on the machine. I updated the dart doc on the global.

Reply:
ahh, ok yeah, that makes sense.